SkyEditor.RomEditor.Rtdx/Domain/Rtdx/Structures/Sir0StringList.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using SkyEditor.IO.Binary;
+using SkyEditor.RomEditor.Domain.Common.Structures;
+
+namespace SkyEditor.RomEditor.Domain.Rtdx.Structures
+{
+    public class Sir0StringList
+    {
+        public List<string> Entries { get; } = new List<string>();
+
+        private Encoding encoding;
+
+        public Sir0StringList(Encoding encoding)
+        {
+            this.encoding = encoding;
+        }
+
+        public Sir0StringList(byte[] data) : this(data, Encoding.Unicode)
+        {
+        }
+        
+        public Sir0StringList(byte[] data, Encoding encoding)
+        {
+            this.encoding = encoding;
+            
+            // IReadOnlyBinaryDataAccessorExtensions.ReadNullTerminatedString() doesn't work with multibyte strings
+            // since the calculated length is only half the length of the actual string.
+            // This workaround is also technically incorrect since it doesn't handle the length of UTF-16 4-byte chars correctly.
+            // It shouldn't be a problem in practise with the files we're reading here though.
+            string ReadNullTerminatedStringMultibyteCompatible(IReadOnlyBinaryDataAccessor accessor, long index)
+            {
+                // The null character we're looking for
+                var nullCharSequence = encoding.GetBytes("\0");
+
+                // Find the length of the string as determined by the location of the null-char sequence
+                int length = 0;
+                while (!accessor.ReadArray(index + length * nullCharSequence.Length, nullCharSequence.Length).All(x => x == 0))
+                {
+                    length += 1;
+                }
+
+                // Multiply the number of characters with the null sequence length to calculate a length that works in most cases
+                return accessor.ReadString(index, length * nullCharSequence.Length, encoding);
+            }
+            
+            var sir0 = new Sir0(data);
+
+            // The first few pointer offsets don't matter here
+            for (int i = 3; i < sir0.PointerOffsets.Count; i++)
```

Comment:
I think this is the result of the first entry going away in the tests. In the header of SIR0 files, there's a pointer to header and a pointer to the pointers section. By starting at 3, you're also skipping the 3rd pointer at index 2.

Tbh I never thought to read SIR0 files using PointerOffsets. That's mainly there to allow the game translate each pointer to a memory address. I personally recommend reading the files like this:
```
var entryCount = sir0.SubHeader.ReadInt32(8);
for (int i = 0; i < entryCount; i++)
{
    long stringOffset = sir0.SubHeader.ReadInt32(16 + i * sizeof(long));
    string value = ReadNullTerminatedStringMultibyteCompatible(sir0.Data, stringOffset);
    Entries.Add(value);
}
```

Reply:
Cool thanks, this is working. Had to read the entryCount at offset 0 though.